Makefile

```diff
@@ -138,34 +138,14 @@ generate-clientset: .remove-clientset .remove-deepcopy .remove-crds
 .remove-clientset:
 	rm -rf $(CURDIR)/$(PKG_BASE)/clientset
 
-# Generate an image for containerized builds
-# Note: This image is local only
-.PHONY: .build-image .pull-build-image .push-build-image
-.build-image: docker/Dockerfile.devel
-	if [ x"$(SKIP_IMAGE_BUILD)" = x"" ]; then \
-		$(DOCKER) build \
-			--progress=plain \
-			--build-arg GOLANG_VERSION="$(GOLANG_VERSION)" \
-			--tag $(BUILDIMAGE) \
-			-f $(^) \
-			docker; \
-	fi
-
-.pull-build-image:
-	$(DOCKER) pull $(BUILDIMAGE)
-
-.push-build-image:
-	$(DOCKER) push $(BUILDIMAGE)
-
-$(DOCKER_TARGETS): docker-%: .build-image
-	@echo "Running 'make $(*)' in docker container $(BUILDIMAGE)"
+$(DOCKER_TARGETS): docker-%: 
+	@echo "Running 'make $(*)' in container image $(BUILDIMAGE)"
 	$(DOCKER) run \
 		--rm \
-		-e HOME=$(PWD) \
-		-e GOCACHE=$(PWD)/.cache/go \
-		-e GOPATH=$(PWD)/.cache/gopath \
-		-v $(PWD):$(PWD) \
-		-w $(PWD) \
+		-e GOCACHE=/tmp/.gocache \
+		-v $(PWD):/work \
+	    -v $(GOMODCACHE):/go/pkg/mod \
```

Comment:
Nit: Update the indent to match the surrounding code.

Also, what is the impact of this if the golang version in the container is different to the host? Should this also be `/tmp/go/pkg/mod` or something?

Reply:
Note that the `.shell` target needs to be updated to match this target too.